examples/test-exec-doc.yml

```diff
@@ -0,0 +1,153 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  pull_request:
+    paths:
+      - '**/*.md'           # Trigger on pull requests affecting Markdown files
+  workflow_dispatch:        # Allows manual triggering of the workflow
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest  # Run the job on the latest Ubuntu runner
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0      # Fetch all history for all branches and tags
+
+    - name: Azure CLI Login
+      uses: azure/login@v1
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}             # Azure Client ID from secrets
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}             # Azure Tenant ID from secrets
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }} # Azure Subscription ID from secrets
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x' # Specify Python version 3.x
```

Comment:
Could you pin this to a specific version?

Reply:
the action works with this line. I tested it

Reply:
Yeah, I believe this uses the latest python and if there are newer versions which are incompatible either with the script or the packages that you use in the script, it will break. It's generally always better to pin to an exact version of what you're going to use for reproducibility as well.